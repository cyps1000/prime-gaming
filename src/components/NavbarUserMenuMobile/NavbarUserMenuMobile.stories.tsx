/**
 * Storybook
 * @see https://storybook.js.org/docs/react/get-started/introduction
 */
import { Story, Meta } from "@storybook/react";

/**
 * Component Imports
 */
import NavbarUserMenuMobile, { NavbarUserMenuMobileProps } from "./NavbarUserMenuMobile";

export default {
  title: "Components/NavbarUserMenuMobile",
  component: NavbarUserMenuMobile,
} as Meta;

/**
 * Defines the Template
 * @param args NavbarUserMenuMobileProps
 * @returns
 */
const Template: Story<NavbarUserMenuMobileProps> = (args) => (
  <NavbarUserMenuMobile {...args} />
);

/**
 * Default case
 */
export const Default = Template.bind({});

/**
 * Put your component props in here
 *
 * Example:
 * Default.args = {
 *  text: "Hello World"
 * }
 *
 * Assuming that the component expects a prop text that is a string.
 */
Default.args = {};
