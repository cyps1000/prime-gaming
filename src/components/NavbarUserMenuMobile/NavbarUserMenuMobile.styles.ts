/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles, fade } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  listItem: {
    color: theme.palette.secondary.main,
    cursor: "pointer",
  },
}));

export { useStyles };
