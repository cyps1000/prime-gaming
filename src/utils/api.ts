import axios from "axios";

/**
 * Defines the hook props interface
 */
interface Config {
  mock: boolean;
}

export const getApiClient = (config: Config) => {
  const { mock } = config;

  /**
   * Handles getting the base api url
   */
  const getApiUrl = () => {
    if (process.env.NODE_ENV === "development") {
      if (mock) return "/";
      return process.env.REACT_APP_LOCAL_API;
    }
    return process.env.REACT_APP_STAGING_API;
  };

  const apiClient = axios.create({
    baseURL: getApiUrl(),
  });

  return { apiClient };
};
