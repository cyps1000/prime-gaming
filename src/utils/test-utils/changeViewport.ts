/**
 * External imports
 */
import mediaQuery from "css-mediaquery";

/**
 * Defines the BreakpointKey type
 */
type BreakpointKey = "xl" | "lg" | "md" | "sm" | "xs";

/**
 * Handles getting the width based on the breakpoint key provided
 */
const getWidthByThemeBreakpoints = (breakpointKey: BreakpointKey) => {
  switch (breakpointKey) {
    case "xl":
      return 1920;
    case "lg":
      return 1280;
    case "md":
      return 960;
    case "sm":
      return 600;
    case "xs":
      return 414;
    default:
      return 1920;
  }
};

/**
 * Handles changing the viewport
 */
export const changeViewport = (breakpointKey: BreakpointKey): any => {
  let width: number = getWidthByThemeBreakpoints(breakpointKey);

  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width: width }),
      addListener: () => {},
      removeListener: () => {},
    };
  };
};
