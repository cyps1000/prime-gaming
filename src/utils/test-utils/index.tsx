import React, { ReactElement } from "react";
/**
 * External Imports
 */
import { MemoryRouter } from "react-router-dom";

/**
 * Imports testing utils
 */
import { render, RenderOptions } from "@testing-library/react";

/**
 * Imports providers
 */

import { ThemeProvider, LanguageProvider } from "../../hooks";
import Providers from "../../components/Providers";

/**
 * Defines the Wrapper component
 */
const Wrapper: React.FC = ({ children }) => {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <Providers>
          <MemoryRouter>{children}</MemoryRouter>
        </Providers>
      </LanguageProvider>
    </ThemeProvider>
  );
};

/**
 * Defines the custom render
 */
const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

/**
 * Re-exports everything from the testing library
 */
export * from "@testing-library/react";

export * from "./changeViewport";
export * from "./mocks";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
