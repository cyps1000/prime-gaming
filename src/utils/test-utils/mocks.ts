export const mockHistoryPush = jest.fn();
