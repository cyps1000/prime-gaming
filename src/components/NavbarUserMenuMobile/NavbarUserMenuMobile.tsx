/**
 * Modules Imports
 */
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";

/**
 * Imports Material UI Components
 */
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";

/**
 * Imports the component styles
 */
import { useStyles } from "./NavbarUserMenuMobile.styles";

/**
 * Defines the props interface
 */
export interface NavbarUserMenuMobileProps {
  text?: string;
  close: () => void;
}

/**
 * Displays the component
 */
const NavbarUserMenuMobile: React.FC<NavbarUserMenuMobileProps> = (props) => {
  const { text, close } = props;

  /**
   * Handles the translations
   */
  const { t } = useTranslation();

  /**
   * Gets the component styles
   */
  const classes = useStyles();

  /**
   * Gets the history object
   */
  const history = useHistory();

  /**
   * Handles Routing to Home Page
   */
  const goToHome = () => {
    history.push("/");
    close();
  };

  return (
    <List className={classes.listItem}>
      <ListItem button>
        <ListItemText onClick={goToHome} primary={t("home")} />
      </ListItem>
      <ListItem button>
        <ListItemText primary={t("news")} />
      </ListItem>
      <ListItem button>
        <ListItemText primary={t("about")} />
      </ListItem>
      <ListItem button>
        <ListItemText primary={t("contact")} />
      </ListItem>
      <ListItem button>
        <ListItemText primary={t("signIn")} />
      </ListItem>
      <ListItem button>
        <ListItemText primary={t("signUp")} />
      </ListItem>
    </List>
  );
};

export default NavbarUserMenuMobile;
