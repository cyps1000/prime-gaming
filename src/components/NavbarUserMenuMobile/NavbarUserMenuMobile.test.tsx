/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 * @see https://www.smashingmagazine.com/2020/07/react-apps-testing-library/
 */

/**
 * Imports test utils
 */
import { render, fireEvent } from "../../utils/test-utils";
import { mockHistoryPush } from "../../utils/test-utils/mocks";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import NavbarUserMenuMobile from "./NavbarUserMenuMobile";

/**
 * NavbarUserMenuMobile rendering tests
 */
describe("NavbarUserMenuMobile Rendering Tests", () => {
  it("renders the component without errors", () => {
    const closeDrawer = jest.fn();
    render(<NavbarUserMenuMobile close={closeDrawer} />);
  });
});

/**
 * NavbarUserMenuMobile logic tests
 */
describe("NavbarUserMenuMobile Logic Tests", () => {
  it("takes the user to the home page when clicked", () => {
    const closeDrawer = jest.fn();
    const { getByText } = render(<NavbarUserMenuMobile close={closeDrawer} />);

    fireEvent.click(getByText(global.en_en["home"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/");
    expect(closeDrawer).toHaveBeenCalled();

    fireEvent.click(getByText(global.en_en["news"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/news");
    expect(closeDrawer).toHaveBeenCalled();

    fireEvent.click(getByText(global.en_en["about"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/about");
    expect(closeDrawer).toHaveBeenCalled();

    fireEvent.click(getByText(global.en_en["contact"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/contact");
    expect(closeDrawer).toHaveBeenCalled();

    expect(mockHistoryPush).toHaveBeenCalledTimes(4);
    expect(closeDrawer).toHaveBeenCalledTimes(4);
  });
});

/**
 * NavbarUserMenuMobile snapshot test
 */
describe("NavbarUserMenuMobile Snapshot Test", () => {
  it("passes the snapshot test", () => {
    const closeDrawer = jest.fn();
    const { container } = render(<NavbarUserMenuMobile close={closeDrawer} />);

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
