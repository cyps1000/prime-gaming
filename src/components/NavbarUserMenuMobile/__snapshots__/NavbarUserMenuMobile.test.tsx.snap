// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarUserMenuMobile Snapshot Test passes the snapshot test 1`] = `
"<ul class=\\"MuiList-root makeStyles-listItem-3 MuiList-padding\\">
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Home</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">News</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">About</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Contact</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Sign In</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
  <div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">
    <div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Sign Up</span></div><span class=\\"MuiTouchRipple-root\\"></span>
  </div>
</ul>"
`;
